starlite/handlers/base.py

```diff
@@ -287,6 +299,13 @@ def _validate_dependency_is_unique(dependencies: dict[str, Provide], key: str, p
                     f"If you wish to override a provider, it must have the same key."
                 )
 
+    def on_startup(self) -> None:
```

Comment:
I think this should be called `on_registration` or some other name the reflects the point at which it is called. We already have "startup" event, which could lead to confusion, this this method will actually be called within `Starlite.register`, and not on application startup or init (it *may* be called on init, but only incidentally).

Reply:
Yes I was uneasy about this too, `on_registration` a much better option.